rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/490_date_histogram_timezone.yml

```diff
@@ -0,0 +1,138 @@
+              date:
+                type: date
+                format: "yyyy-MM-dd HH:mm:ss"
+
+  - do:
+      bulk:
+        index: test
+        refresh: true
+        body:
+          - { "index": {} }
+          - { "date": "2021-05-01 20:00:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 21:30:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 23:54:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:45:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 23:40:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:20:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 21:20:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 23:50:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:55:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:15:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 21:10:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 23:35:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:40:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:20:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:10:00" }
+          - { "index": {} }
+          - { "date": "2021-05-01 22:25:00" }
+
+---
+"date_histogram and date_histogram_composite timezone":
+  - skip:
+      version: " - 8.1.99"
+      reason: bug fixed in 8.2.0
+  - do:
+      search:
+        body:
+          size: 0
+          aggs:
+            date_histogram_with_tz:
+              date_histogram:
+                field: date
+                format: "yyyy-MM-dd HH:mm:ss"
+                calendar_interval: hour
+                time_zone: Asia/Jakarta
+            date_histogram_without_tz:
+              date_histogram:
+                field: date
+                format: "yyyy-MM-dd HH:mm:ss"
+                calendar_interval: hour
+            date_histogram_composite_with_tz:
+              composite:
+                size: 20
+                sources:
+                  - date_field:
+                      date_histogram:
+                        field: date
+                        format: "yyyy-MM-dd HH:mm:ss"
+                        calendar_interval: hour
+                        time_zone: Asia/Jakarta
+            date_histogram_composite_without_tz:
+              composite:
+                size: 20
+                sources:
+                  - date_field:
+                      date_histogram:
+                        field: date
+                        format: "yyyy-MM-dd HH:mm:ss"
+                        calendar_interval: hour
+
+  - match: { hits.total.value: 16 }
+  - match: { hits.total.relation: "eq" }
+  - length: { aggregations.date_histogram_with_tz.buckets: 4 }
+  - match: { aggregations.date_histogram_with_tz.buckets.0.key_as_string: "2021-05-02 03:00:00" }
```

Comment:
I wonder if you can make the names the same width and check the two bucket starts side by side so it's really easy to see the time zone effects. Like:
```
   - match: { aggregations.yes_tz.buckets.0.key_as_string: "2021-05-02 03:00:00" }
   - match: { aggregations.no__tz.buckets.0.key_as_string: "2021-05-02 04:00:00" }
```